internal/backend/db_mailbox.go

```diff
@@ -166,7 +166,7 @@ func DBGetMailboxMessageIDPairs(ctx context.Context, client *ent.Client, mailbox
 	}), nil
 }
 
-func DBGetUIDInterval(ctx context.Context, mbox *ent.Mailbox, begin, end int) ([]*ent.UID, error) {
+func DBGetUIDInterval(ctx context.Context, client *ent.Client, mbox *ent.Mailbox, begin, end int) ([]*ent.UID, error) {
```

Comment:
need client here?

Reply:
While we don't need to use the client object, it is there to enforce the DB type use and to make sure we have acquire the lock. Otherwise, these queries will run at any moment, bypassing the lock and causing the infamous "database locked" error.